src/System.Private.CoreLib/shared/System/Type.cs

```diff
@@ -346,6 +346,21 @@ public virtual Array GetEnumValues()
         public virtual Type MakeGenericType(params Type[] typeArguments) { throw new NotSupportedException(SR.NotSupported_SubclassOverride); }
         public virtual Type MakePointerType() { throw new NotSupportedException(); }
 
+        public static Type MakeGenericSignatureType(Type genericTypeDefinition, params Type[] typeArguments)
+        {
+            if (genericTypeDefinition == null)
+                throw new ArgumentNullException(nameof(genericTypeDefinition));
+            if (typeArguments == null)
+                throw new ArgumentNullException(nameof(typeArguments));
+            for (int i = 0; i < typeArguments.Length; i++)
+            {
+                if (typeArguments[i] == null)
```

Comment:
It would be a bit more robust to validate this after cloning.

Reply:
Yes, the validation here is intentionally minimal as the main use for this api is for MakeGenericType implementations which have already done the cloning.

Reply:
> MakeGenericType implementations which have already done the cloning

Depends on the implementation. It is not required for MakeGenericType to clone the instantiation to call this. E.g. `RtType.MakeGenericType` does not clone the array before passing it to SignatureConstructedGenericType constructor.

What I meant is that the validation could be in SignatureConstructedGenericType after the array is cloned.

Reply:
Hmm... that was an annoying oversight. Ok, replacing the debug.assert in the constructor with actual validation.